handshake/src/types/txin.rs

```diff
@@ -0,0 +1,146 @@
+
+use std::io::{Read, Write};
+
+use riemann_core::{
+    hashes::marked::MarkedDigest,
+    ser::{ByteFormat, SerError, SerResult},
+    types::tx::{Input, TXOIdentifier},
+};
+
+use rmn_btc::{hashes::TXID, types::txin::{Outpoint}};
+
+/// A simple type alias for an outpoint type that will be repeated throught the `bitcoin` module.
+pub type HandshakeOutpoint = Outpoint<TXID>;
+
+// TODO: I need to change one public function on Outpoint
+// from_explorer_format should not swap the endianness,
+// because logic makes sense.
```

Comment:
Did handshake ecosystem change the endianness junk?

Reply:
Yes the handshake ecosystem uses the same endianness everywhere instead of flipping it for UI

Reply:
I decided to take the code for the `Outpoint` and pull it into this crate and remove the methods that aren't needed. Maybe the `Outpoint` could go in `coins_core`?

Reply:
I'm leaning towards adding the `witness` to the Handshake `Outpoint` and removing the `witnesses` vector from the `HandshakeTx` struct. This will simplify serialization logic when there is no witness present as it will default to the 0 varint